### PR TITLE
Add example for tracks

### PR DESCRIPTION
**Pull request by Rodrigo-Tenorio**
Let's have a common place to generate spectrogram plots containing CW signals, shall we?

### CODE REVIEW

File: examples/other_examples/spectrogram_example.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import matplotlib.pyplot as plt
+
+import pyfstat
+
+plt.rcParams["font.family"] = "serif"
+plt.rcParams["font.size"] = 18
+plt.rcParams["text.usetex"] = True
+
+label = "track_example"
+outdir = os.path.join(sys.path[0], "tmp")
```
Comment (dbkeitel) [inline examples/other_examples/spectrogram_example.py:14]:
```suggestion
```

File: examples/other_examples/spectrogram_example.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    "h0": data_parameters["sqrtSX"] / depth,
+    "cosi": 1.0,
+}
+
+data = pyfstat.BinaryModulatedWriter(
+    label=label, outdir=outdir, **data_parameters, **signal_parameters
+)
+data.make_data()
+
+times, freqs, sft_data = pyfstat.helper_functions.get_sft_array(data.sftfilepath)
```
Comment (dbkeitel) [inline examples/other_examples/spectrogram_example.py:45]:
```suggestion
with tempfile.TemporaryDirectory() as tmpdir:
    data = pyfstat.BinaryModulatedWriter(
        label=label, outdir=tmpdir, **data_parameters, **signal_parameters
    )
   data.make_data()

   times, freqs, sft_data = pyfstat.helper_functions.get_sft_array(data.sftfilepath)
```

File: examples/other_examples/spectrogram_example.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    (times - times[0]) / 86400,
+    freqs,
+    (sft_data / data_parameters["sqrtSX"])[:-1, :-1],
+    cmap="inferno_r",
+)
+fig.colorbar(c, label="Normalized Power")
+plt.tight_layout()
+fig.savefig("spectrogram_example.png")
+
+shutil.rmtree(outdir)
```
Comment (dbkeitel) [inline examples/other_examples/spectrogram_example.py:60]:
```suggestion
```

File: examples/other_examples/spectrogram_example.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+times, freqs, sft_data = pyfstat.helper_functions.get_sft_array(data.sftfilepath)
+
+fig, ax = plt.subplots(figsize=(0.8 * 16, 0.8 * 9))
+ax.grid(which="both")
+ax.set(xlabel="Time [days]", ylabel="Frequency [Hz]", ylim=(99.98, 100.02))
+c = ax.pcolormesh(
+    (times - times[0]) / 86400,
+    freqs,
+    (sft_data / data_parameters["sqrtSX"])[:-1, :-1],
```
Comment (dbkeitel) [inline examples/other_examples/spectrogram_example.py:53]: It doesn't really matter for an illustrative example plot, but I think _technically_ here you're dropping one data point (SFT) and the "proper" solution would be to add end points to the `times` and `freqs` instead...?
(Just recently ran into this annoying "n+1,n+1,n*n" convention of `pcolormesh` elsewhere...)

File: pyfstat/helper_functions.py [hunk 5]
```diff
@@ -357,7 +357,7 @@ def get_sft_array(sftfilepattern, data_duration=None, F0=None, dF0=None):
         )
 
     SFTs = MultiSFTs.data[0]
-    times = [sft.epoch.gpsSeconds for sft in SFTs.data]
+    times = np.array([sft.epoch.gpsSeconds for sft in SFTs.data])
```
Comment (dbkeitel) [inline pyfstat/helper_functions.py:360]: Good catch, it really made no sense to have `data` numpyfied but `times` not.

File: tests.py [hunk 6]
```diff
@@ -555,8 +555,8 @@ def test_glitch_injection(self):
         max_freq_noglitch = freqs_noglitch[np.argmax(data_noglitch, axis=0)]
         max_freq_glitch = freqs_glitch[np.argmax(data_glitch, axis=0)]
         print([max_freq_vanilla, max_freq_noglitch, max_freq_glitch])
-        self.assertEqual(times_noglitch, times_vanilla)
-        self.assertEqual(times_glitch, times_vanilla)
+        self.assertTrue(np.allclose(times_noglitch, times_vanilla))
+        self.assertTrue(np.allclose(times_glitch, times_vanilla))
```
Comment (dbkeitel) [inline tests.py:559]: Our times still are integers, aren't they? So shouldn't we still check for exact equality? I guess you could either cast it back to a list for `assertEqual` to digest, or there's also an `numpy.testing.assert_array_equal`.

File: examples/other_examples/spectrogram_example.py (hunk 4 above)
Comment (Rodrigo-Tenorio) [reply examples/other_examples/spectrogram_example.py:53]: Yes. I'll think about this.

File: tests.py (hunk 6 above)
Comment (Rodrigo-Tenorio) [reply tests.py:559]: That's true. I'll look into `numpy.testing.assert_array_equal`

File: examples/other_examples/spectrogram_example.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
+import os
+import shutil
```
Comment (Rodrigo-Tenorio) [inline examples/other_examples/spectrogram_example.py:2]:
```suggestion
```

File: examples/other_examples/spectrogram_example.py (hunk 4 above)
Comment (Rodrigo-Tenorio) [reply examples/other_examples/spectrogram_example.py:50]: `shading="nearest"` does exactly what I intended to do!

File: tests.py (hunk 6 above)
Comment (Rodrigo-Tenorio) [reply tests.py]: Why not just `np.all(a == b)`, since `a` and `b` are integer arrays.

Comment (Rodrigo-Tenorio) [reply tests.py]: a65eafb

**Review by dbkeitel — APPROVED** [review]

**Pull request merged**